src/server/src/modules/user.ts

```diff
@@ -104,10 +104,54 @@ export const getUserByEmail = async (userId: number, email: string) => {
         message: '',
     };
 
+    const receiver = await findUserByEmail(email)
+
+    // Check if the user with the email exists.
+    if (!receiver) {
+        getUserByEmailResult.message =
+            'A user with the email does not exist.';
+        return getUserByEmailResult;
+    }
+
+    if (receiver.id == userId) {
+        getUserByEmailResult.message =
+            'You cannot send message to yourself.';
+        return getUserByEmailResult;
+    }
+
+    getUserByEmailResult.message = 'Successful';
+
+    const getUserByIdResult = await getUserById(receiver.id)
```

Comment:
Do you need to getUserById? I thought the receiver object would already store a user object.

Reply:
Because user object also returns encoded password, which you said we don't want to return it.

Reply:
Because the user object also contains an encoded password, which you said we don't want to return to the client. I will make my code more readable.

Reply:
oh, I see. But since findUserByEmail returns to you a user object, I think you can grab the properties you need from that object and then return as needed. Something like:
`const { firstName, lastName, id } = findUserByEmail(email)`

Reply:
Done.